lib/streamaggr/rate.go

```diff
@@ -29,6 +29,9 @@ func getRateAggrSharedValue(isGreen bool) *rateAggrSharedValue {
 var rateAggrStateValuePool sync.Pool
 
 func putRateAggrStateValue(v *rateAggrStateValue) {
+	if v == nil {
```

Comment:
@AndrewChubatiuk can we add a test case that could have caught this?
I know that test stream aggregation is not an easy, however if not checking the correctness of the output  we still can verify races and issues like this with unittests.

Reply:
updated method, so now this error reproducible in tests